packages/concerto-core/test/model/concept.js

```diff
@@ -128,6 +128,17 @@ describe('Concept', function () {
             }).should.throw(/Attempting to create an ENUM declaration is not supported./);
         });
 
+        it('should generate an error trying to create an Map from JSON', function () {
+            let conceptModel = fs.readFileSync('./test/data/model/concept.cto', 'utf8');
+            modelManager.addCTOModel(conceptModel, 'concept.cto');
+            const factory = new Factory(modelManager);
+            const serializer = new Serializer(factory, modelManager);
+            const jsObject = JSON.parse('{"$class":"org.acme.biznet.Dictionary"}');
+            (function () {
+                serializer.fromJSON(jsObject);
+            }).should.throw(/Attempting to create an Map declaration is not supported./);
```

Comment:
Will this be supported in future, or is there some more fundamental issue?

Reply:
I'll answer with a question—does it make sense to serialize a MapDeclaration independently of a concept?

Reply:
No, but we should be able to for a MapDeclaration within a Concept.